ltrace-powerpc-Define-_GNU_SOURCE-to-enable-strndup-for-uClibc.patch

For ltrace on powerpc, we need to #define _GNU_SOURCE when using older
glibc or uClibc before including <string.h> to enable the strndup()
function.
---
 ...-_GNU_SOURCE-to-enable-strndup-for-uClibc.patch |   27 ++++++++++++++++++++
 1 file changed, 27 insertions(+)
 create mode 100644 patches/ltrace/0.7.3/200-Define-_GNU_SOURCE-to-enable-strndup-for-uClibc.patch

diff --git a/patches/ltrace/0.7.3/200-Define-_GNU_SOURCE-to-enable-strndup-for-uClibc.patch b/patches/ltrace/0.7.3/200-Define-_GNU_SOURCE-to-enable-strndup-for-uClibc.patch
new file mode 100644
index 0000000..e6e0202
--- /dev/null
+++ b/patches/ltrace/0.7.3/200-Define-_GNU_SOURCE-to-enable-strndup-for-uClibc.patch
@@ -0,0 +1,27 @@
+From 81b9fd9cf37fffbf12c8a406e26611d4230d20da Mon Sep 17 00:00:00 2001
+From: Curt Brune <curt@cumulusnetworks.com>
+Date: Thu, 6 Aug 2015 08:26:51 -0700
+Subject: [PATCH] ppc:  Define _GNU_SOURCE to enable strndup() for uClibc
+
+When using older glibc or uClibc, we need to #define _GNU_SOURCE
+before including <string.h> to enable the strndup() function
+---
+ sysdeps/linux-gnu/ppc/plt.c | 2 ++
+ 1 file changed, 2 insertions(+)
+
+diff --git a/sysdeps/linux-gnu/ppc/plt.c b/sysdeps/linux-gnu/ppc/plt.c
+index 5f81889..09a0887 100644
+--- a/sysdeps/linux-gnu/ppc/plt.c
++++ b/sysdeps/linux-gnu/ppc/plt.c
+@@ -20,6 +20,8 @@
+  * 02110-1301 USA
+  */
+ 
++#define _GNU_SOURCE
++
+ #include <gelf.h>
+ #include <sys/ptrace.h>
+ #include <errno.h>
+-- 
+1.9.1
+
